src/clients/scopes/DecicatedScope.tsx

```diff
@@ -0,0 +1,156 @@
+import React, { useState } from "react";
+import { useTranslation } from "react-i18next";
+import {
+  AlertVariant,
+  Divider,
+  FormGroup,
+  PageSection,
+  Switch,
+} from "@patternfly/react-core";
+
+import type ClientRepresentation from "@keycloak/keycloak-admin-client/lib/defs/clientRepresentation";
+import { FormAccess } from "../../components/form-access/FormAccess";
+import { HelpItem } from "../../components/help-enabler/HelpItem";
+import { useAdminClient } from "../../context/auth/AdminClient";
+import { useAlerts } from "../../components/alert/Alerts";
+import {
+  mapRoles,
+  RoleMapping,
+  Row,
+} from "../../components/role-mapping/RoleMapping";
+import type { RoleMappingPayload } from "@keycloak/keycloak-admin-client/lib/defs/roleRepresentation";
+
+type DedicatedScopeProps = {
+  client: ClientRepresentation;
+};
+
+export const DedicatedScope = ({
+  client: initialClient,
+}: DedicatedScopeProps) => {
+  const { t } = useTranslation("clients");
+  const adminClient = useAdminClient();
+  const { addAlert, addError } = useAlerts();
+
+  const [client, setClient] = useState<ClientRepresentation>(initialClient);
+  const [hide, setHide] = useState(false);
+
+  const loader = async () => {
+    const [assignedRoles, effectiveRoles] = await Promise.all([
+      adminClient.clients
+        .listRealmScopeMappings({ id: client.id! })
+        .then((roles) => roles.map((role) => ({ role }))),
+      adminClient.clients
+        .listCompositeRealmScopeMappings({ id: client.id! })
+        .then((roles) => roles.map((role) => ({ role }))),
+    ]);
+
+    const clients = await adminClient.clients.find();
```

Comment:
I think you can also get the clients in parallel here.
```suggestion
    const [clients, assignedRoles, effectiveRoles] = await Promise.all([
      adminClient.clients.find(),
      adminClient.clients
        .listRealmScopeMappings({ id: client.id! })
        .then((roles) => roles.map((role) => ({ role }))),
      adminClient.clients
        .listCompositeRealmScopeMappings({ id: client.id! })
        .then((roles) => roles.map((role) => ({ role }))),
    ]);

```